crnlib/Makefile

```diff
@@ -1,7 +1,12 @@
 .DEFAULT_GOAL := all
 .PHONY: all clean
 
-COMPILE_OPTIONS = -O3 -fomit-frame-pointer -ffast-math -fno-math-errno -g -fno-strict-aliasing -Wall -Wno-unused-value -Wno-unused -march=core2
+CXX = g++
+COMPILE_OPTIONS = $(COMPILE_CPU_OPTIONS) $(COMPILE_DEBUG_OPTIONS) $(COMPILE_OPTIMIZATION_OPTIONS) $(COMPILE_WARN_OPTIONS)
```

Comment:
Are you sure this line must not be _after_ the others, like this?

```make
COMPILE_CPU_OPTIONS = -march=core2
COMPILE_DEBUG_OPTIONS = -g
COMPILE_OPTIMIZATION_OPTIONS = -O3 -fomit-frame-pointer -ffast-math -fno-math-errno -fno-strict-aliasing
COMPILE_WARN_OPTIONS = -Wall -Wno-unused-value -Wno-unused
COMPILE_OPTIONS = $(COMPILE_CPU_OPTIONS) $(COMPILE_DEBUG_OPTIONS) $(COMPILE_OPTIMIZATION_OPTIONS) $(COMPILE_WARN_OPTIONS)
```

Reply:
Ordering of declarations in Makefiles doesn't matter, both work just fine.

Reply:
Oh yes, `COMPILE_OPTIONS` is not evaluated there.

Still, would it be possible to put the COMPILE_OPTIONS _after_, just to prevent misinterpretation for free? I know this is bikeshedding. :grin: